### PR TITLE
[cpu][bench] Add Fused MoE Micro Benchmark for CPU Backend

### PR DESCRIPTION
**Pull request by andikarachman**
## Purpose

Add Fused MoE Micro Benchmark for CPU Backend
Fixes: #31721 

## Test Plan
Tested using this instance:
Screenshot 2026-01-10 at 22 51 33

## Test Result
Screenshot 2026-01-10 at 22 53 15
---
<details>

<summary> Essential Elements of an Effective PR Description Checklist </summary>

- [x] The purpose of the PR, such as "Fix some issue (link existing issues this PR will resolve)".
- [x] The test plan, such as providing test command.
- [x] The test results, such as pasting the results comparison before and after, or e2e results
- [ ] (Optional) The necessary documentation update, such as updating `supported_models.md` and `examples` for a new model.
- [ ] (Optional) Release notes update. If your change is user facing, please update the release notes draft in the [Google Doc](https://docs.google.com/document/d/1YyVqrgX4gHTtrstbq8oWUImOyPCKSGnJ7xtTpmXzlRs/edit?tab=t.0).
</details>

---

> [!NOTE]
> Adds a CPU fused MoE micro-benchmark.
> 
> - Introduces `benchmarks/kernels/cpu/benchmark_cpu_fused_moe.py` to time `cpu_fused_moe` with prepacked weights (`cpu_prepack_moe_weight`) and report latency stats and TFLOP/s
> - Supports ISA selection (`amx`/`vec`), activation choice (`silu`/`swigluoai`), optional bias, reproducible seeding, and configurable shapes/top-k/iters via CLI
> - Validates CPU ops availability and exits with a clear message if missing; includes warmup runs before measurement
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit b0c40f26c84326672211b5815abd16a87a2c361b. This will update automatically on new commits. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>

### CODE REVIEW

File: benchmarks/kernels/cpu/benchmark_cpu_fused_moe.py [hunk 1]
```diff
@@ -0,0 +1,192 @@
...
+
+import os
+import sys
+import time
+
+import numpy as np
+import torch
+
+# Check VLLM_TARGET_DEVICE before importing vllm modules
+if os.environ.get("VLLM_TARGET_DEVICE", "").lower() != "cpu":
```
Comment (fadara01) [inline benchmarks/kernels/cpu/benchmark_cpu_fused_moe.py]: The benchmark is under a CPU folder which implies that it's CPU specific.
We also don't need to have `VLLM_TARGET_DEVICE` set to `cpu` as these benchmarks are only relevant on a CPU-only machine IMO.

File: benchmarks/kernels/cpu/benchmark_cpu_fused_moe.py [hunk 2]
```diff
@@ -0,0 +1,192 @@
...
+    args = parser.parse_args()
+
+    # Default topk_num to expert_num // 2, minimum 1
+    topk_num = (
+        args.topk_num if args.topk_num is not None else max(args.expert_num // 2, 1)
+    )
+
+    print(args)
+
+    print(
```
Comment (fadara01) [inline benchmarks/kernels/cpu/benchmark_cpu_fused_moe.py]: Isn't this redundant if you're printing the args above?

**Review by fadara01 — CHANGES_REQUESTED** [review]
Thanks for doing this!
LGTM, just some minor nits

File: benchmarks/kernels/cpu/benchmark_cpu_fused_moe.py (hunk 1 above)
Comment (andikarachman) [reply benchmarks/kernels/cpu/benchmark_cpu_fused_moe.py]: you are right,  I have deleted this part

File: benchmarks/kernels/cpu/benchmark_cpu_fused_moe.py (hunk 2 above)
Comment (andikarachman) [reply benchmarks/kernels/cpu/benchmark_cpu_fused_moe.py]: my bad, I have removed the redundant part

**Review by bigPYJ1151 — APPROVED** [review]
Thanks! LGTM

**Review by fadara01 — APPROVED** [review]
Thanks for addressing the comments :) 
LGTM!
